diff -Naur Urho3D-1.8/cmake/Modules/CheckCompilerToolchain.cmake b/cmake/Modules/CheckCompilerToolchain.cmake
--- Urho3D-1.8/cmake/Modules/CheckCompilerToolchain.cmake	2022-01-24 10:41:12.000000000 +1100
+++ b/cmake/Modules/CheckCompilerToolchain.cmake	2024-07-07 20:10:24.649955514 +1000
@@ -226,7 +226,7 @@
     elseif (POWERPC)
         check_extension (altivec)
     elseif (X86 OR E2K)
-        foreach (ext sse sse2 sse3 sse4 avx avx2)
+        foreach (ext sse sse2 sse3 sse4)
             check_extension (${ext})
         endforeach ()
         if (CMAKE_SYSTEM_NAME STREQUAL Linux)
diff -Naur Urho3D-1.8/cmake/Modules/UrhoCommon.cmake b/cmake/Modules/UrhoCommon.cmake
--- Urho3D-1.8/cmake/Modules/UrhoCommon.cmake	2022-01-24 10:41:12.000000000 +1100
+++ b/cmake/Modules/UrhoCommon.cmake	2024-07-07 20:10:24.649955514 +1000
@@ -639,6 +639,9 @@
                     set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=${URHO3D_DEPLOYMENT_TARGET}")
                 endif ()
             endif ()
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mno-avx -mno-avx2 -mno-avx512f")
+            set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -mno-avx -mno-avx2 -mno-avx512f")
+
             # We don't add these flags directly here for Xcode because we support Mach-O universal binary build
             # The compiler flags will be added later conditionally when the effective arch is i386 during build time (using XCODE_ATTRIBUTE target property)
             if (NOT XCODE)
diff -Naur Urho3D-1.8/CMakeLists.txt b/CMakeLists.txt
--- Urho3D-1.8/CMakeLists.txt	2022-01-24 10:41:12.000000000 +1100
+++ b/CMakeLists.txt	2024-07-07 20:10:24.649955514 +1000
@@ -25,6 +25,8 @@
 
 # Set project name
 project (Urho3D)
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mno-avx -mno-avx2 -mno-avx512f")
+set(CMAKE_CXX_FLAGS "${CMAKE_C_FLAGS} -mno-avx -mno-avx2 -mno-avx512f")
 
 # Set CMake modules search path
 set (CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/cmake/Modules)
diff -Naur Urho3D-1.8/Source/ThirdParty/Bullet/src/LinearMath/btQuickprof.cpp b/Source/ThirdParty/Bullet/src/LinearMath/btQuickprof.cpp
--- Urho3D-1.8/Source/ThirdParty/Bullet/src/LinearMath/btQuickprof.cpp	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/Bullet/src/LinearMath/btQuickprof.cpp	2024-07-07 20:12:35.246844121 +1000
@@ -43,7 +43,7 @@
 #else  //_XBOX
 #include <windows.h>
 
-#if WINVER < 0x0602
+#if 1
 #define GetTickCount64 GetTickCount
 #endif
 
diff -Naur Urho3D-1.8/Source/ThirdParty/LibCpuId/src/cpuid_main.c b/Source/ThirdParty/LibCpuId/src/cpuid_main.c
--- Urho3D-1.8/Source/ThirdParty/LibCpuId/src/cpuid_main.c	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/LibCpuId/src/cpuid_main.c	2024-07-07 20:10:24.649955514 +1000
@@ -196,13 +196,13 @@
 		{ 25, CPU_FEATURE_AES },
 		{ 26, CPU_FEATURE_XSAVE },
 		{ 27, CPU_FEATURE_OSXSAVE },
-		{ 28, CPU_FEATURE_AVX },
+//		{ 28, CPU_FEATURE_AVX },
 		{ 29, CPU_FEATURE_F16C },
 		{ 30, CPU_FEATURE_RDRAND },
 	};
 	const struct feature_map_t matchtable_ebx7[] = {
 		{  3, CPU_FEATURE_BMI1 },
-		{  5, CPU_FEATURE_AVX2 },
+//		{  5, CPU_FEATURE_AVX2 },
 		{  8, CPU_FEATURE_BMI2 },
 	};
 	const struct feature_map_t matchtable_edx81[] = {
@@ -592,7 +592,7 @@
 		{ CPU_FEATURE_AES, "aes" },
 		{ CPU_FEATURE_XSAVE, "xsave" },
 		{ CPU_FEATURE_OSXSAVE, "osxsave" },
-		{ CPU_FEATURE_AVX, "avx" },
+//		{ CPU_FEATURE_AVX, "avx" },
 		{ CPU_FEATURE_MMXEXT, "mmxext" },
 		{ CPU_FEATURE_3DNOW, "3dnow" },
 		{ CPU_FEATURE_3DNOWEXT, "3dnowext" },
@@ -631,19 +631,19 @@
 		{ CPU_FEATURE_APERFMPERF, "aperfmperf" },
 		{ CPU_FEATURE_PFI, "pfi" },
 		{ CPU_FEATURE_PA, "pa" },
-		{ CPU_FEATURE_AVX2, "avx2" },
+//		{ CPU_FEATURE_AVX2, "avx2" },
 		{ CPU_FEATURE_BMI1, "bmi1" },
 		{ CPU_FEATURE_BMI2, "bmi2" },
 		{ CPU_FEATURE_HLE, "hle" },
 		{ CPU_FEATURE_RTM, "rtm" },
-		{ CPU_FEATURE_AVX512F, "avx512f" },
-		{ CPU_FEATURE_AVX512DQ, "avx512dq" },
-		{ CPU_FEATURE_AVX512PF, "avx512pf" },
-		{ CPU_FEATURE_AVX512ER, "avx512er" },
-		{ CPU_FEATURE_AVX512CD, "avx512cd" },
+//		{ CPU_FEATURE_AVX512F, "avx512f" },
+//		{ CPU_FEATURE_AVX512DQ, "avx512dq" },
+//		{ CPU_FEATURE_AVX512PF, "avx512pf" },
+//		{ CPU_FEATURE_AVX512ER, "avx512er" },
+//		{ CPU_FEATURE_AVX512CD, "avx512cd" },
 		{ CPU_FEATURE_SHA_NI, "sha_ni" },
-		{ CPU_FEATURE_AVX512BW, "avx512bw" },
-		{ CPU_FEATURE_AVX512VL, "avx512vl" },
+//		{ CPU_FEATURE_AVX512BW, "avx512bw" },
+//		{ CPU_FEATURE_AVX512VL, "avx512vl" },
 		{ CPU_FEATURE_SGX, "sgx" },
 		{ CPU_FEATURE_RDSEED, "rdseed" },
 		{ CPU_FEATURE_ADX, "adx" },
diff -Naur Urho3D-1.8/Source/ThirdParty/LibCpuId/src/recog_intel.c b/Source/ThirdParty/LibCpuId/src/recog_intel.c
--- Urho3D-1.8/Source/ThirdParty/LibCpuId/src/recog_intel.c	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/LibCpuId/src/recog_intel.c	2024-07-07 20:10:24.649955514 +1000
@@ -376,16 +376,16 @@
 		{  2, CPU_FEATURE_SGX },
 		{  4, CPU_FEATURE_HLE },
 		{ 11, CPU_FEATURE_RTM },
-		{ 16, CPU_FEATURE_AVX512F },
-		{ 17, CPU_FEATURE_AVX512DQ },
+//		{ 16, CPU_FEATURE_AVX512F },
+//		{ 17, CPU_FEATURE_AVX512DQ },
 		{ 18, CPU_FEATURE_RDSEED },
 		{ 19, CPU_FEATURE_ADX },
-		{ 26, CPU_FEATURE_AVX512PF },
-		{ 27, CPU_FEATURE_AVX512ER },
-		{ 28, CPU_FEATURE_AVX512CD },
+//		{ 26, CPU_FEATURE_AVX512PF },
+//		{ 27, CPU_FEATURE_AVX512ER },
+//		{ 28, CPU_FEATURE_AVX512CD },
 		{ 29, CPU_FEATURE_SHA_NI },
-		{ 30, CPU_FEATURE_AVX512BW },
-		{ 31, CPU_FEATURE_AVX512VL },
+//		{ 30, CPU_FEATURE_AVX512BW },
+//		{ 31, CPU_FEATURE_AVX512VL },
 	};
 	if (raw->basic_cpuid[0][0] >= 1) {
 		match_features(matchtable_edx1, COUNT_OF(matchtable_edx1), raw->basic_cpuid[1][3], data);
diff -Naur Urho3D-1.8/Source/ThirdParty/SDL/src/cpuinfo/SDL_cpuinfo.c b/Source/ThirdParty/SDL/src/cpuinfo/SDL_cpuinfo.c
--- Urho3D-1.8/Source/ThirdParty/SDL/src/cpuinfo/SDL_cpuinfo.c	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/SDL/src/cpuinfo/SDL_cpuinfo.c	2024-07-07 20:10:24.649955514 +1000
@@ -659,6 +659,7 @@
             SDL_CPUFeatures |= CPU_HAS_SSE42;
             SDL_SIMDAlignment = SDL_max(SDL_SIMDAlignment, 16);
         }
+/*
         if (CPU_haveAVX()) {
             SDL_CPUFeatures |= CPU_HAS_AVX;
             SDL_SIMDAlignment = SDL_max(SDL_SIMDAlignment, 32);
@@ -671,6 +672,7 @@
             SDL_CPUFeatures |= CPU_HAS_AVX512F;
             SDL_SIMDAlignment = SDL_max(SDL_SIMDAlignment, 64);
         }
+*/
         if (CPU_haveNEON()) {
             SDL_CPUFeatures |= CPU_HAS_NEON;
             SDL_SIMDAlignment = SDL_max(SDL_SIMDAlignment, 16);
diff -Naur Urho3D-1.8/Source/ThirdParty/SLikeNet/CMakeLists.txt b/Source/ThirdParty/SLikeNet/CMakeLists.txt
--- Urho3D-1.8/Source/ThirdParty/SLikeNet/CMakeLists.txt	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/SLikeNet/CMakeLists.txt	2024-07-07 20:10:24.649955514 +1000
@@ -36,16 +36,26 @@
 endif()
 
 # Check specific function availability which may be missing from different MinGW versions
+# We OVERRIDE the detection result for sprintf_s and vsnprintf_s below with the result of this
+# because Windows XP does not have any of these functions in its msvcrt.dll regardless of them
+# being present in the supposedly windows XP compliant version of MinGW
+check_cxx_source_compiles ("#include <stdio.h>\n#include <time.h>\nint main() { struct tm s;\ntime_t epoch = 0;\nint e = localtime_s(&s, epoch);\nreturn e; }" LOCALTIMES_FUNCTION_EXISTS)
+
+# Check specific function availability which may be missing from different MinGW versions
 check_cxx_source_compiles ("#include <stdio.h>\nint main() { char s[80];\nint x = 1;\nsprintf_s(s, 80, \"Test:%d\", x); }" SPRINTFS_FUNCTION_EXISTS)
 
-if (NOT SPRINTFS_FUNCTION_EXISTS)
+if (NOT LOCALTIMES_FUNCTION_EXISTS)
     add_definitions(-DSPRINTFS_FUNCTION_MISSING=1)
+    add_definitions(-D_INC_STRING_S=1)
+    add_definitions(-D_INC_STDIO_S=1)
+    add_definitions(-D_INC_STDLIB_S=1)
+    add_definitions(-D_INC_WCHAR_S=1)
 endif()
 
 # Check specific function availability which may be missing from different MinGW versions
 check_cxx_source_compiles ("#include <stdio.h>\nint main() {\nchar s[80];\nint x = 1;\nva_list args = {};\nvsnprintf_s(s, 80, \"Test:%d\", args);}" VSNPRINTFS_FUNCTION_EXISTS)
 
-if (NOT VSNPRINTFS_FUNCTION_EXISTS)
+if (NOT LOCALTIMES_FUNCTION_EXISTS)
     add_definitions(-DVSNPRINTFS_FUNCTION_MISSING=1)
 endif()
 
diff -Naur Urho3D-1.8/Source/ThirdParty/SLikeNet/Source/include/slikenet/linux_adapter.h b/Source/ThirdParty/SLikeNet/Source/include/slikenet/linux_adapter.h
--- Urho3D-1.8/Source/ThirdParty/SLikeNet/Source/include/slikenet/linux_adapter.h	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/SLikeNet/Source/include/slikenet/linux_adapter.h	2024-07-07 20:10:24.649955514 +1000
@@ -40,9 +40,7 @@
 
 #ifdef VSNPRINTFS_FUNCTION_MISSING
 int vsnprintf_s(char *buffer, size_t sizeOfBuffer, size_t count, const char *format, va_list argptr);
-#endif
 
-#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
 template<size_t BufferSize> int vsnprintf_s(char (&buffer)[BufferSize], size_t count, const char *format, va_list argptr)
 {
     return vsnprintf_s(buffer, BufferSize, count, format, argptr);
diff -Naur Urho3D-1.8/Source/ThirdParty/SQLite/src/sqlite3.c b/Source/ThirdParty/SQLite/src/sqlite3.c
--- Urho3D-1.8/Source/ThirdParty/SQLite/src/sqlite3.c	2022-01-24 10:41:12.000000000 +1100
+++ b/Source/ThirdParty/SQLite/src/sqlite3.c	2024-07-07 20:12:01.446613471 +1000
@@ -43430,7 +43430,7 @@
 #define osLoadPackagedLibrary ((HMODULE(WINAPI*)(LPCWSTR, \
         DWORD))aSyscall[69].pCurrent)
 
-#if SQLITE_OS_WINRT
+#if 0
   { "GetTickCount64",          (SYSCALL)GetTickCount64,          0 },
 #else
   { "GetTickCount64",          (SYSCALL)0,                       0 },
